### PR TITLE
Issue #4722 - remove websocket-servlet

### PR DESCRIPTION
**Pull request by lachlan-roberts**
**Issue #4722**

Removed the `websocket-servlet` layer which was resulting in exposing `websocket-core` classes.

- `WebSocketServlet` and `WebSocketServletFactory` have been merged into `JettyWebSocketServlet` and `JettyWebSocketServletFactory` which were extending them previously.

- The other classes including `WebSocketUpgradeFilter` and `WebSocketMapping` were moved into the new package `org.eclipse.jetty.websocket.util.server` of `websocket-util`, but maybe these should be in `websocket-core` instead

### CODE REVIEW

**Comment by lachlan-roberts** [comment]
I'm getting a failure on this branch for `DistributionTests.testSimpleWebAppWithWebsocket` saying 
```
java.util.ServiceConfigurationError: org.slf4j.spi.SLF4JServiceProvider: Provider org.eclipse.jetty.logging.JettyLoggingServiceProvider not found
```
Any idea what would cause this @joakime  @olamy , is this something to do with the recent logging changes?

**Comment by olamy** [comment]
@lachlan-roberts should be fixed with this commit https://github.com/eclipse/jetty.project/commit/5199e78587276962028d39b5c5ddfa16a963b8f9

**Review by joakime — APPROVED** [review]
:+1:

**Review by joakime — APPROVED** [review]
Still a :+1:  from me

**Review by olamy — APPROVED** [review]

**Comment by lachlan-roberts** [comment]
@joakime the change to make `jetty-servlet` an optional dependency for `websocket-util` has caused some problems.

The last commit got it working working but giving a JPMS warning as there are servlet classes in the public signature of `websocket-util` server package.
```
Warning:(106, 19) java: class org.eclipse.jetty.servlet.FilterHolder in module org.eclipse.jetty.servlet is not indirectly exported using requires transitive
```

As an alternative I could introduce another module which is shared only between both `websocket-jetty-server` and `websocket-javax-server`, called something like `websocket-server-common` and then move the `WebSocketUpgradeFilter` and other classes here. I still think `WebSocketServlet` does not belong in that package and the current changes to merge that with `JettyWebSocketServlet` is a good change.

**Comment by gregw** [comment]
@lachlan-roberts I'm loath to introduce more modules.  I'm sure JPMS can deal with optional/provided dependencies.... ask @sbordet for advice or look for other examples of provided dependencies in the code base.

**Comment by lachlan-roberts** [comment]
@gregw I asked @sbordet for advice and he said it would be best to introduce a new `websocket-util-server` module.

**Comment by lachlan-roberts** [comment]
I pushed a commit to introduce a `websocket-util-server` module which has the `WebSocketUpgradeFilter` and other classes in internal package.

It only exports the internal package to `websocket-jetty-server` and `websocket-javax-server` and exports the base package (containing only the upgrade filter) to all.

**Comment by joakime** [comment]
The osgi tests on this branch are failing.

File: jetty-osgi/pom.xml [hunk 1]
```diff
@@ -90,7 +90,7 @@
       </dependency>
       <dependency>
         <groupId>org.eclipse.jetty</groupId>
-        <artifactId>jetty-websocket</artifactId>
+        <artifactId>jetty-websocket</artifactId> <!-- TODO: is this even right? -->
```
Comment (lachlan-roberts) [inline jetty-osgi/pom.xml]: @janbartel while debugging I saw this, any idea what this is for? I don't know what this is referring to, I don't think there is any `jetty-websocket` artifactId in jetty 10 so I thought this should be an errror.

**Comment by lachlan-roberts** [comment]
@gregw || @joakime can I get a tick after the new changes where I added the module `jetty-util-server`? I'd like to get this merged before we do another release of 10.

**Review by gregw — APPROVED** [review]
LGTM.

**Pull request merged**